### DISCUSSION TITLE
Kernel 5.15.x LTS

### DISCUSSION THREAD

**Discussion by bluerapik** [AdityaGarg8/mbp-16.x-ubuntu-kernel#94] [General]
Hi Guys.

2 years ago I cloned this project, and in the LTS branch, in the build.sh script, the Kernel version was:

`KERNEL_VERSION=5.15.77`

A few days ago, based on this clone of the project that I have locally, I was able to compile the **5.15.161** LTS kernel (the last LTS of version 5.15.x so far), with some changes in some patches from the repository <https://github.com/t2linux/linux-t2-patches.git>

If I clone the project today, I cannot find the kernel version **5.15.77**, in the build.sh script. Looking at the tags, the last one that theoretically refers to kernel **5.15.x** is the tag: **v5.15.94-1** and the kernel version in the build.sh script is:

`KERNEL_VERSION=6.1.12`

My question is:

1. What happened to the commits in the LTS branch for version **5.15.x**? since I can't find them.

2. And if I want to continue compiling future releases of the **5.15.x** LTS kernel, and I clone this project today, which branch can I base it on?

Thank you for your work on this project!

**Comment by AdityaGarg8** [comment]
> Hi Guys.
> 
> 2 years ago I cloned this project, and in the LTS branch, in the build.sh script, the Kernel version was:
> 
> `KERNEL_VERSION=5.15.77`
> 
> A few days ago, based on this clone of the project that I have locally, I was able to compile the **5.15.161** LTS kernel (the last LTS of version 5.15.x so far), with some changes in some patches from the repository https://github.com/t2linux/linux-t2-patches.git
> 
> If I clone the project today, I cannot find the kernel version **5.15.77**, in the build.sh script. Looking at the tags, the last one that theoretically refers to kernel **5.15.x** is the tag: **v5.15.94-1** and the kernel version in the build.sh script is:
> 
> `KERNEL_VERSION=6.1.12`
> 
> My question is:
> 
> 1. What happened to the commits in the LTS branch for version **5.15.x**? since I can't find them.

Whenever a new LTS is released, we abandon the older LTS completely. Although now I've improved my habit to keeping commit histories, but back in time when 6.1 became the LTS, I deleted the LTS branch and copied the Mainline branch, which hosted 6.1 to LTS.

So the possibility of getting the code that used to run 5.15 would be in the Mainline branch.

> 2. And if I want to continue compiling future releases of the **5.15.x** LTS kernel, and I clone this project today, which branch can I base it on?
> 

You need to keep your local project tbh + update the patches yourself. As said before, we have abandoned older LTS.

> Thank you for your work on this project!

**Comment by AdityaGarg8** [comment]
As far as tags are concerned, looks like I'll need to use a separate action to correctly make tags

**Reply by AdityaGarg8** [reply]
Issue of tags being incorrect associated with their commits on LTS is indeed an issue I never noticed. Cannot fix this for older releases, but I've pushed change to the CI to assign correct tags in the future.

**Comment by bluerapik** [comment]
Thanks @AdityaGarg8 . Then I am going to create a fork of this project to continue with the LTS that still have security support, such as **v5.15.x**, and that will no longer be in this repository.
